src/rosdep2/lookup.py

```diff
@@ -127,8 +127,16 @@ def get_rule_for_platform(self, os_name, os_version, installer_keys, default_ins
                     # we've already checked for PACKAGE_MANAGER_KEY, so
                     # version key must be present here for data to be valid
                     # dictionary value.
-                    if os_version not in data:
+                    if (
```

Comment:
IMO it'd be even more clarifying for the users if different error message returns per a) explicitly set `null` v.s. b) N/A-not-defined in rosdep rule, as the reaction users may need to take differs. For (a) users may either give up using rosdep for that dependency or seek source, and for (b) users may add the missing key to the rosdep rule if the rule is simply missing in rule list.

Reply:
👍

Reply:
Good point. I separated the conditions and raised two different messages in 4b0685d4623a442194f8f7cc97336733791d16ed.